.github/workflows/deployment_tests.yml

```diff
@@ -0,0 +1,247 @@
+
+      # Setup main.tf, deploy and destroy k8s
+      - name: deploy k8s
+        run: |
+          cd k8s/terraform
+          sed -i "s/gcs-bucket-name/${{ steps.vars.outputs.GCP_PROJECT_ID }}_bucket/" main.tf
+          sed -i "s/source-path/..\/..\/..\/..\/..\/mcp/" main.tf
+          cat main.tf
+          terraform init
+          terraform plan -out="./plan.tfplan"
+          terraform apply plan.tfplan
+          terraform plan -destroy -out="./destroy.tfplan"
+          terraform apply destroy.tfplan
+
+      # Deploy app image to Artifact Repository
+      - name: Build App
+        run: |
+          cd GCP/app/build/helloworld
+          gcloud artifacts repositories create app-repo --repository-format=docker --location=europe-west2 --description="Repository for storing app"
+          gcloud builds submit --tag europe-west2-docker.pkg.dev/${{ steps.vars.outputs.GCP_PROJECT_ID }}/app-repo/helloworld
+
+      # Configure yaml files to contain relevant project info
+      - name: Add image URI to GCP Files
+        run: |
+          cd GCP
+          sed -i "s/<image_uri>/europe-west2-docker.pkg.dev\/${{ steps.vars.outputs.GCP_PROJECT_ID }}\/app-repo\/helloworld:latest/" gcp_ae.yml
+          sed -i "s/<image_uri>/europe-west2-docker.pkg.dev\/${{ steps.vars.outputs.GCP_PROJECT_ID }}\/app-repo\/helloworld:latest/" gcp_cloudrun.yml
+
+      # Setup main.tf and deploy gcp modules. Also plan destroy but don't apply due to app engine issues.
+      - name: Deploy GCP modules
+        run: |
+          cd GCP/terraform
+          sleep 5m
+          sed -i "s/gcs-bucket-name/${{ steps.vars.outputs.GCP_PROJECT_ID }}_bucket/" main.tf
+          sed -i "s/source-path/..\/..\/..\/..\/..\/mcp/" main.tf
+          cat main.tf
+          terraform init
+          terraform plan -out="./plan.tfplan"
+          terraform apply plan.tfplan
+          terraform plan -destroy -out="./destroy.tfplan"
+
+      - name: delete project
+        if : ${{ always() }}
+        run: |
+          gcloud projects delete ${{ steps.vars.outputs.GCP_PROJECT_ID }}
+          gcloud beta billing projects unlink ${{ steps.vars.outputs.GCP_PROJECT_ID }}
+
+  ############################# Deployment Tests for GCP Access Context Manager ####################################
+  ## Job to check if secret with service account for deploying perimeter exists
+  check-vpc-SA:
+    runs-on: ubuntu-latest
+    outputs:
+      sa-exists: ${{ steps.check.outputs.EXISTS }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check service account exists
+        id: check
+        run: |
+          if [[ "${{ secrets.GCP_VPC_SA_KEY }}" == *"private_key_id"* ]]; then
+            echo ::set-output name=EXISTS::true
+          else
+            echo ::set-output name=EXISTS::false
+          fi
+
+  # If service account exists run this job
+  GCP-perimeter-tests:
+    runs-on: ubuntu-latest
+    needs: check-vpc-sa
```

Comment:
Is this case sensitive?

Reply:
Yup it is sorry. Changed job name to `check-vpc-sa` to match those